### PR TITLE
pcre2: update to 10.40

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by neheb** [comment]
Adorable error on MSYS2...

```
Test 3: Locale-specific features (using 'french' locale)
** Locale test did not run successfully. The output did not match
   test3output, test3outputA or test3outputB.
   This may mean that there is a problem with the locale settings rather
   than a bug in PCRE2.
```

**Comment by neheb** [comment]
fails with the regular MSYS2 pcre2 package. I'll just ignore.

There's also a RunGrepTest. Wonder if that should be added...

**Comment by neheb** [comment]
Amazing..
sshot-17
.

**Comment by neheb** [comment]
@jpakkane ready

File: ci_config.json [hunk 1]
```diff
@@ -254,6 +254,12 @@
       "libboost-iostreams-dev"
     ]
   },
+  "pcre2": {
+    "_comment": "Tests are broken on Windows",
+    "build_options": [
+      "pcre2:test=false"
```
Comment (eli-schwartz) [inline ci_config.json:260]: NACK

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: ci_config.json (hunk 1 above)
Comment (eli-schwartz) [reply ci_config.json:260]: > There's also a RunGrepTest. Wonder if that should be added...

You added another test, the test that was already there works, but the one you added fails on Windows. The solution to this isn't to then disable all tests for all platforms.

Simply mark this one as `should_fail: true` on that platform.

Comment (neheb) [reply ci_config.json]: I think this should do it.

File: subprojects/packagefiles/pcre2/meson.build [hunk 2]
```diff
@@ -160,13 +160,25 @@ if get_option('test') and not meson.is_cross_build() # wine wrappers are not bat
...
   test('RunTest', runtest,
+    should_fail: host_machine.system() == 'windows' and c_compiler.get_argument_syntax() != 'msvc',
+    env : ['srcdir=@0@'.format(meson.current_source_dir())],
+    workdir: meson.current_build_dir())
+
+  if c_compiler.get_argument_syntax() == 'msvc'
+    rungreptest = find_program('RunGrepTest.bat')
+  else
+    rungreptest = find_program('RunGrepTest')
+  endif
```
Comment (eli-schwartz) [inline subprojects/packagefiles/pcre2/meson.build:178]: TBH I'd put that in the same if/else that finds `runtest`.

File: ci_config.json (hunk 1 above)
Comment (eli-schwartz) [reply ci_config.json]: Yep, thanks. :)

File: subprojects/packagefiles/pcre2/meson.build (hunk 2 above)
Comment (neheb) [reply subprojects/packagefiles/pcre2/meson.build:178]: you mean

```
if get_option('test') and not meson.is_cross_build()
```

?

Comment (eli-schwartz) [reply subprojects/packagefiles/pcre2/meson.build:178]:
```meson
  if c_compiler.get_argument_syntax() == 'msvc'
    runtest = find_program('RunTest.bat')
  else
    runtest = find_program('RunTest')
  endif


  if c_compiler.get_argument_syntax() == 'msvc'
    rungreptest = find_program('RunGrepTest.bat')
  else
    rungreptest = find_program('RunGrepTest')
  endif
```

Why not just
```meson
  if c_compiler.get_argument_syntax() == 'msvc'
    runtest = find_program('RunTest.bat')
    rungreptest = find_program('RunGrepTest.bat')
  else
    runtest = find_program('RunTest')
    rungreptest = find_program('RunGrepTest')
  endif
```

**Comment by neheb** [comment]
Anyway, I reported the test failures upstream.

**Comment by eli-schwartz** [comment]
Great. For the sake of my future self, that seems to be https://github.com/PCRE2Project/pcre2/issues/127

**Pull request merged**